partners_api/partners_api_tests/booking_tests.cpp

```diff
@@ -6,33 +6,51 @@
 
 #include "base/scope_guard.hpp"
 
+#include <chrono>
+
 using namespace partners_api;
+using namespace booking;
+using namespace booking::http;
 
 namespace
 {
 UNIT_TEST(Booking_GetHotelAvailability)
 {
   string const kHotelId = "98251";  // Booking hotel id for testing.
   string result;
-  TEST(booking::RawApi::GetHotelAvailability(kHotelId, "", result), ());
+  TEST(RawApi::GetHotelAvailability(kHotelId, "", result), ());
   TEST(!result.empty(), ());
 }
 
 UNIT_TEST(Booking_GetExtendedInfo)
 {
   string const kHotelId = "0";  // Internal hotel id for testing.
   string result;
-  TEST(booking::RawApi::GetExtendedInfo(kHotelId, "en", result), ());
+  TEST(RawApi::GetExtendedInfo(kHotelId, "en", result), ());
   TEST(!result.empty(), ());
 }
 
+UNIT_TEST(Booking_HotelAvailability)
+{
+  AvailabilityParams params;
+  params.m_hotelIds = {"98251"};
+  params.m_rooms = {"A,A"};
+  params.m_checkin = std::chrono::system_clock::now() + std::chrono::hours(24);
+  params.m_checkout = std::chrono::system_clock::now() + std::chrono::hours(24 * 7);
+  params.m_stars = {"4", "5"};
+  string result;
+  TEST(RawApi::HotelAvailability(params, result), ());
+  TEST(!result.empty(), ());
+  LOG(LINFO, (result));
+}
+
 UNIT_CLASS_TEST(AsyncGuiThread, Booking_GetMinPrice)
 {
-  booking::SetBookingUrlForTesting("http://localhost:34568/booking/min_price");
-  MY_SCOPE_GUARD(cleanup, []() { booking::SetBookingUrlForTesting(""); });
+  SetBookingUrlForTesting("http://localhost:34568/booking/min_price");
```

Comment:
It would be nice to subclass AsyncGuiThread and do this stuff in ctor and dtor:
```c++
class BookingApiTests
{
public:
  BookingApiTest() { SetBookingUrlForTesting(...); }
  ~BookingApiTest() { SetBookingUrlForTesting(""); }
};
```
BTW, do you really need AsyncGuiThread here? Why?

Reply:
сделал наследника. использую AsyncGuiThread потому, что в качестве коллбэков использую SafeCallback